spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/actuate/AbstractGatewayControllerEndpoint.java

```diff
@@ -93,11 +97,28 @@ public void setApplicationEventPublisher(ApplicationEventPublisher publisher) {
 	// TODO: Add uncommited or new but not active routes endpoint
 
 	@PostMapping("/refresh")
-	public Mono<Void> refresh() {
-		this.publisher.publishEvent(new RefreshRoutesEvent(this));
+	public Mono<Void> refresh(@RequestParam(value = "metadata", required = false) List<String> byMetadata) {
+		publishRefreshEvent(byMetadata);
```

Comment:
Can you describe the use case for this?

Is it useful in the case when there are a large number of routes and you only updated a subset, so you will see some performance impact by refreshing everything?

Reply:
Absolutely!

`RouteDefinition` elements are stored in SCG independently using the `/actuator/gateway/routes/{id}` or the batch endpoint. 
Currently, if a developer runs the endpoint `/actuator/gateway/refresh` all the `RouteDefinition` elements are processed and the Routes are created and updated for the RouteLocator. 

Two main use cases
* A developer wants to refresh routes as a group, so if one fails, it only impacts the new routes coming from that group
* Avoid creating routes again for a portion of the route definitions

Example
Having in SCG: RouteDefinitions = [] and Routes = []

POST `/actuator/gateway/routes/route-1-1` with metadata.groupBy=group-1
POST `/actuator/gateway/routes/route-2-1` with metadata.groupBy=group-2
POST `/actuator/gateway/routes/route-1-2` with metadata.groupBy=group-1

> RouteDefinitions = [route-1-1, route-2-1, route-1-2]  

`/refresh?metadata=groupBy=group-1` only refresh routes whose metadata contains groupBy=group-1: `route-1-1` and `route-1-2`

Outcome Routes = **[ `route-1-1`, `route-1-2`]**
Instead of _[route-1-1, route-2-1, route-1-2]_  

For a large set of RouteDefinitions (1000) but we only refresh 5 routes, it will avoid creating 995 new Route objects (saving time also initiating routes)

Reply:
Well for me I was wondering if the two changes there related to each other..if one required the other.  I also could make an argument that if we had the revert one change or the other we wouldn't need to revert both (even though one didn't need to be reverted).

I am not going to say you need to separate the two for this PR, just explaining why I was pointing this out